Nil pointer dereference on startup

### Describe the bug

On Prysm 7.1.3.

```
[2026-05-21 17:04:45.51]  INFO p2p: Updating ENR Fork ID CurrentForkDigest=0x8c9f62fe NextForkDigest=0x8c9f62fe NextForkEpoch=18446744073709551615 NextForkVersion=0x06000000
[2026-05-21 17:04:45.64]  INFO p2p: Started discovery v5 ENR=enr:-Ni4QJ4ZwiD27G-GJWMNa1F2madv1zRgaFP02uNJ_Wgx7_F0WOv_o2D5BC1zmvZ_XM3O0o03jeJ71-qsfphvC92lOTqGAZsKH3aoh2F0dG5ldHOIAAAAAAAAAACDY2djCIRldGgykIyfYv4GAAAA__________-CaWSCdjSCaXCErBIAAoNuZmSEAAAAAIRxdWljgiMpiXNlY3AyNTZrMaEDRuv0DQZA5i4QXcKt2lA-bYRiiv1uPh5ClTJuvqbKLu-Ic3luY25ldHMAg3RjcIIyyIN1ZHCCLuA seq=1765401392808
[2026-05-21 17:04:45.79]  INFO p2p: New node record ENR=enr:-Ni4QCw2PUaTDcnOnJYIWCHGzSQ6HEFFvGtemLvHkKl3N9vGHA7ZABQV_cAkh7nCJ9lOz3vvOfV2ocHV86hMgl6c_WyGAZsKH3aph2F0dG5ldHOIAAwAAAAAAACDY2djCIRldGgykIyfYv4GAAAA__________-CaWSCdjSCaXCErBIAAoNuZmSEAAAAAIRxdWljgiMpiXNlY3AyNTZrMaEDRuv0DQZA5i4QXcKt2lA-bYRiiv1uPh5ClTJuvqbKLu-Ic3luY25ldHMAg3RjcIIyyIN1ZHCCLuA seq=1765401392809
[2026-05-21 17:04:45.79]  INFO p2p: Node started p2p server multiAddr=/ip4/172.18.0.2/tcp/13000/p2p/16Uiu2HAmHRqZtHiFaLv5ophfMAEU4FrfLnmxYYgNs7uy5WWgSaaS
[2026-05-21 17:04:45.79]  INFO p2p: Node started p2p server multiAddr=/ip4/172.18.0.2/udp/9001/quic-v1/p2p/16Uiu2HAmHRqZtHiFaLv5ophfMAEU4FrfLnmxYYgNs7uy5WWgSaaS
[2026-05-21 17:04:47.00]  INFO p2p: Updating ENR Fork ID CurrentForkDigest=0x8c9f62fe NextForkDigest=0x8c9f62fe NextForkEpoch=18446744073709551615 NextForkVersion=0x06000000
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x46db959]

goroutine 92899 [running]:
github.com/OffchainLabs/prysm/v7/beacon-chain/sync/initial-sync.(*Service).fetchOriginSidecars(0xc06041e000, {0xc0c10f20c0, 0x3, 0x3})
        beacon-chain/sync/initial-sync/service.go:242 +0x119
github.com/OffchainLabs/prysm/v7/beacon-chain/sync/initial-sync.(*Service).Start(0xc06041e000)
        beacon-chain/sync/initial-sync/service.go:213 +0x10eb
created by github.com/OffchainLabs/prysm/v7/runtime.(*ServiceRegistry).StartAll in goroutine 1
        runtime/service_registry.go:45 +0x35
Checkpoint sync enabled
MEV Boost enabled
No MEV Build Factor configured: Prysm doesn't support it, or Eth Docker doesn't know how.
Prysm node with beacon DB pruning
```

### Has this worked before in a previous version?

```text
Yes, Been running Prysm for years without issue.
```

### 🔬 Minimal Reproduction

start prysm
### Error



### Platform(s)

docker (eth-docker project)

### What version of Prysm are you running? (Which release)

7.1.3

--- a/beacon-chain/db/kv/blocks.go
+++ b/beacon-chain/db/kv/blocks.go
@@ -477,14 +477,10 @@ func (s *Store) DeleteHistoricalDataBeforeSlot(ctx context.Context, cutoffSlot p
 		return 0, err
 	}
 
-	// Return early if there's nothing to delete.
-	if len(slotRoots) == 0 {
-		return 0, nil
-	}
-
 	// Perform all deletions in a single transaction for atomicity
 	var numSlotsDeleted int
 	err = s.db.Update(func(tx *bolt.Tx) error {
+		blocksBkt := tx.Bucket(blocksBucket)
 		for _, sr := range slotRoots {
 			// Return if context is cancelled or deadline is exceeded.
 			if ctx.Err() != nil {
@@ -541,6 +537,13 @@ func (s *Store) DeleteHistoricalDataBeforeSlot(ctx context.Context, cutoffSlot p
 			s.blockCache.Del(string(sr.root[:]))
 			// Delete state summary from cache
 			s.stateSummaryCache.delete(sr.root)
+		}
+
+		originRoot := blocksBkt.Get(originCheckpointBlockRootKey)
+		if originRoot != nil && blocksBkt.Get(originRoot) == nil {
+			if err = blocksBkt.Delete(originCheckpointBlockRootKey); err != nil {
+				return errors.Wrap(err, "could not delete origin checkpoint block root")
+			}
 		}
 
 		return nil

--- a/beacon-chain/db/kv/blocks_test.go
+++ b/beacon-chain/db/kv/blocks_test.go
@@ -618,6 +618,66 @@ func TestStore_HistoricalDataBeforeSlot(t *testing.T) {
 
 }
 
+func TestStore_DeleteHistoricalDataBeforeSlot_ClearsOriginCheckpointPointerWhenOriginBlockPruned(t *testing.T) {
+	db := setupDB(t)
+	ctx := t.Context()
+
+	require.NoError(t, db.SaveGenesisBlockRoot(ctx, genesisBlockRoot))
+	blks := makeBlocks(t, 0, 8, genesisBlockRoot)
+	require.NoError(t, db.SaveBlocks(ctx, blks))
+
+	originBlk := blks[2]
+	originRoot, err := originBlk.Block().HashTreeRoot()
+	require.NoError(t, err)
+	require.NoError(t, db.SaveOriginCheckpointBlockRoot(ctx, originRoot))
+
+	gotRoot, err := db.OriginCheckpointBlockRoot(ctx)
+	require.NoError(t, err)
+	require.Equal(t, originRoot, gotRoot)
+	gotBlock, err := db.Block(ctx, originRoot)
+	require.NoError(t, err)
+	require.NotNil(t, gotBlock)
+
+	slotsDeleted, err := db.DeleteHistoricalDataBeforeSlot(ctx, originBlk.Block().Slot()+1, 8)
+	require.NoError(t, err)
+	require.NotEqual(t, 0, slotsDeleted)
+
+	_, err = db.OriginCheckpointBlockRoot(ctx)
+	require.ErrorIs(t, err, ErrNotFoundOriginBlockRoot)
+	gotBlock, err = db.Block(ctx, originRoot)
+	require.NoError(t, err)
+	require.Equal(t, nil, gotBlock)
+}
+
+func TestStore_DeleteHistoricalDataBeforeSlot_ClearsAlreadyDanglingOriginCheckpointPointer(t *testing.T) {
+	db := setupDB(t)
+	ctx := t.Context()
+
+	require.NoError(t, db.SaveGenesisBlockRoot(ctx, genesisBlockRoot))
+	blks := makeBlocks(t, 0, 2, genesisBlockRoot)
+	require.NoError(t, db.SaveBlocks(ctx, blks))
+
+	originBlk := blks[0]
+	originRoot, err := originBlk.Block().HashTreeRoot()
+	require.NoError(t, err)
+	require.NoError(t, db.SaveOriginCheckpointBlockRoot(ctx, originRoot))
+	require.NoError(t, db.DeleteBlock(ctx, originRoot))
+
+	gotRoot, err := db.OriginCheckpointBlockRoot(ctx)
+	require.NoError(t, err)
+	require.Equal(t, originRoot, gotRoot)
+	gotBlock, err := db.Block(ctx, originRoot)
+	require.NoError(t, err)
+	require.Equal(t, nil, gotBlock)
+
+	slotsDeleted, err := db.DeleteHistoricalDataBeforeSlot(ctx, primitives.Slot(0), 8)
+	require.NoError(t, err)
+	require.Equal(t, 0, slotsDeleted)
+
+	_, err = db.OriginCheckpointBlockRoot(ctx)
+	require.ErrorIs(t, err, ErrNotFoundOriginBlockRoot)
+}
+
 func TestStore_GenesisBlock(t *testing.T) {
 	db := setupDB(t)
 	ctx := t.Context()

--- a/changelog/satushh_clear-dangling-origin-checkpoint-pointer.md
+++ b/changelog/satushh_clear-dangling-origin-checkpoint-pointer.md
@@ -1,0 +1,3 @@
+### Fixed
+
+- Clear the origin checkpoint block root pointer in `DeleteHistoricalDataBeforeSlot` when the origin block has been pruned, so `OriginCheckpointBlockRoot` no longer returns a dangling root.

